crates/transpiler/src/target/mod.rs

```
@@ -511,12 +513,22 @@ impl Target {
     ///             target.instruction_supported("rx", (0,), RXGate, parameters=[pi / 4])
     ///
     ///         will return ``True`` if an RXGate(pi/4) exists on qubit 0.
+    ///     check_angle_bounds (bool): If set to True the value of ``parameters`` will be validated
+    ///         against any angle bounds set in the target. By default this is set to False as in
+    ///         the preset transpiler pipeline checking this is handled directly by the
+    ///         :class:`.GatesInBasis` pass and adjusting to respect angle bounds is handled
+    ///         automatically by the :class:`.WrapAngles` pass so this check would restrict other
+    ///         transpilations passes. However for validating all the instructions in a target or
+    ///         when building custom transpilation pipelines you will want to set this to ``True``.
+    ///         If any of the values in ``parameters`` are set to be :class:`.ParameterExpression`
+    ///         instances this flag will have no effect as angle bounds only impact
+    ///         non-parameterized operations in the circuit.
```

I got into this a lot more in the main review comment, but flagging here: I think the addition of this method defaulting to `false` fundamentally changes `instruction_supported`.  If anything, I think it should default to `true`, and if we ever use it in Qiskit code, we should adjust our code.  We ought to cause passes to fail loudly, not silently produce incorrect output - I think any claim of "backwards compatibility" already comes with a lot of qualifiers, and that this errs on the wrong side in the name of something we're already breaking.

For similar reasons, I think the Rust-space `Target::instruction_supported` should also check the bound angles; I think it's a foot-gun waiting to go off if we don't deal with it now, but silently soften the method to returning "instruction is supported if you do some more translation work".

---

I flipped the default in https://github.com/Qiskit/qiskit/pull/14406/commits/240dab7a24c52fc877c44aa9789418024dad712f 

The rust space method doesn't seem to understand parameters at all!? I think we can leave this for a follow up potentially.